inc/Engine/Admin/PerformanceMonitoring/Database/Queries/PerformanceMonitoring.php

```diff
@@ -157,4 +157,25 @@ public function make_blurred( $row_id ) {
 			]
 		);
 	}
+
+	/**
+	 * Limit rows to keep the passed argument and remove others.
+	 *
+	 * @param int $to_keep Number of rows to keep.
+	 *
+	 * @return void
+	 */
+	public function limit_urls( int $to_keep ) {
+		$ids = $this->query(
+			[
+				'fields'  => 'ids',
+				'offset'  => $to_keep,
+				'orderby' => 'modified',
+				'order'   => 'asc',
+			]
+			);
+		foreach ( $ids as $id ) {
+			$this->delete_item( $id );
+		}
```

Comment:
I think we should optimize this to run the query once, instead of being in a loop and hitting the DB for every id? if not now, then we should open an issue for this and consider an enhancement.